config/info.go

```diff
@@ -0,0 +1,71 @@
+package config
+
+import (
+	"sync"
+
+	"github.com/tbd54566975/ssi-service/pkg/service/framework"
+)
+
+const (
+	ServiceName    = "ssi-service"
+	ServiceVersion = "0.0.3"
+)
+
+var (
+	serviceInfo *ServiceInfo
+	once        sync.Once
+)
+
+// GetServiceInfo provides ServiceInfo as a singleton
+func GetServiceInfo() *ServiceInfo {
+	once.Do(func() {
```

Comment:
I found it surprising to use sync mechanisms only in this function, but not in the underlying object. All the setter methods are not concurrency-safe. In particular, `SetServicePath` may fail if it's called concurrently by multiple go-routines. 

I recommend not using `once`, and keep this simple by exporting `serviceInfo`, as well as all the fields within it.

Reply:
yeah probably better to keep it simple

Reply:
read about its usage here https://thedevelopercafe.com/articles/singleton-in-golang-839d8610958b

the use of `once.Do` enforces that only one caller will obtain a lock on the singleton at a time, so the setter methods are indeed thread safe

Reply:
That applies only to callers of the GetServiceInfo function. The same does not hold for different threads that already have a handle to the `*ServiceInfo` instance, and both calling `SetServicePath` at the same time.

Reply:
It's not a risk outside of the config package, I can move it to its own package if that would remove your concern

Reply:
re-refactored to remove concurrency controls; added a warning